helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -467,7 +467,8 @@ private void chargePendingTransition(Resource resource, CurrentStateOutput curre
       for (Message message : pendingMessages) {
         StateTransitionThrottleConfig.RebalanceType rebalanceType =
             getRebalanceTypePerMessage(requiredStates, message, currentStateMap);
-        String currentState = currentStateMap.get(message.getTgtName());
+        String currentState = currentStateMap.get(message.getTgtName()) == null ? stateModelDefinition.getInitialState()
+            : currentStateMap.get(message.getTgtName());
```

Comment:
nit, how about just add a null check like this, I think it's simpler and sightly more efficient.

String currentState = currentStateMap.get(message.getTgtName());
if (currentState == null) {
  currentState = stateModelDefinition.getInitialState();
}